src/math/matrix.lisp

```diff
@@ -116,7 +116,7 @@
                          (in-vec matrix1)
                          (in-vec matrix2)))))
 
-(defmacro define-matrix-num-op (name matrix-type map-fun)
+(defmacro define-matrix-num-strord-op (name matrix-type map-fun)
   `(defmethod ,name ((matrix ,matrix-type) (value real))
```

Comment:
The body of the methods are exaclty the same, and now we have 3 macros.
What I meant was something like this:
```lisp
(defmacro define-matrix-num-op (name matrix-type map-fun)
  (let ((body `(make-instance ',matrix-type :in-vec
                              (map 'vector (alexandria:curry ,map-fun value)
                                   (in-vec matrix)))))
    `(progn
       (defmethod ,name ((value real) (matrix ,matrix-type))
         ,body)
       (defmethod ,name ((matrix ,matrix-type) (value real))
         ,body))))
```

Reply:
There's actually one major flaw with this: `revord` macros use simple `curry`, while `strord` uses `rcurry`, and I don't see a way of doing it the neat way because of this.